Made windows related dependencies optional

newpavlov opened the pull request.

danburkert commented on README.md:
```diff
@@ -1,7 +1,7 @@
 
 Extended utilities for working with files and filesystems in Rust. `fs2`
-requires Rust stable 1.1 or greater.
+requires Rust stable 1.8 or greater.
```
Could you also bump the travis version?

danburkert commented on Cargo.toml:
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "fs2"
-version = "0.2.5"
+version = "0.3.0"
```
Don't make this change; I'll bump it when I make a new release.

danburkert commented on Cargo.toml:
```diff
@@ -9,7 +9,9 @@ description = "Cross-platform file locks and file duplication."
 
 [dependencies]
-libc = "0.2.2"
+libc = "0.2"
```
I can't recall exactly, but I think there was a reason libc "0.2.1" was incompatible.

newpavlov commented:
Fixed all your comments. In addition to bumping 1.1 to 1.8 in travis config I've also added stable release.

Regarding making `libc` optional instead of using 'cfg(not(windows))' I've used 'cfg(unix)' as it's how you do imports in `lib.rs`.

danburkert commented:
Perfect, thanks!

danburkert commented:
(I'll do a release shortly)
